pkg/controller/disruption/disruption.go

```diff
@@ -433,12 +433,7 @@ func (dc *DisruptionController) getPodsForPdb(pdb *policy.PodDisruptionBudget) (
 	if err != nil {
 		return []*v1.Pod{}, err
 	}
-	// TODO: Do we need to copy here?
-	result := make([]*v1.Pod, 0, len(pods))
-	for i := range pods {
-		result = append(result, &(*pods[i]))
-	}
-	return result, nil
+	return pods, nil
```

Comment:
audit done, on different code paths, callers do not modify pods currently.
I can also add a comment to state that we should not modify returned pods.